### PR TITLE
Databases/week2

### PR DESCRIPTION
**Pull request by WENCHO22**

### CODE REVIEW

File: databases/week2/homework-part-1.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
+USE hyf_lesson1;
+-- 1. Add a task
+INSERT INTO task(title, description, created, updated, due_date, status_id, user_id) VALUES("Clean the kitchen", "Leave it shining", "2020-10-12 10:37:37", "2020-10-12 12:37:37", "2020-12-25 12:37:37", 1, 2);
```
Comment (flemeur) [inline databases/week2/homework-part-1.sql]: Consider splitting long queries up in multiple lines

File: databases/week2/homework-part-1.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+WHERE id = 2;
+
+-- 4. change task status
+UPDATE task
+SET status_id = 3
+WHERE id = 3;
+
+-- 5. mark a task as complete
+UPDATE task
+SET status_id = 4
```
Comment (flemeur) [inline databases/week2/homework-part-1.sql]: Wouldn't status_id = 3 be done/complete?

File: databases/week2/homework-part-1.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+select *
+from task;
+
+
+
+
+
+
+
+
```
Comment (flemeur) [inline databases/week2/homework-part-1.sql]: Remember to remove any irrelevant queries from the homework before handing it in :)

File: databases/week2/homework-part-3.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+WHERE user.name = "Donald Duck" AND status.name = "Done";
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT user.name, task.title, task.created
+FROM user_task
+JOIN task
+ON user_task.task_id = task.id
+JOIN user
+ON user_task.user_id = user.id
+WHERE user.name = "Maryrose Meadows" AND month(task.created) = 9 AND MONTH;
```
Comment (flemeur) [inline databases/week2/homework-part-3.sql]: This query gives me an error: `Error Code: 1054. Unknown column 'MONTH' in 'where clause'`

File: databases/week2/homework-part-3.sql [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+-- Find how many tasks where created in each month, e.g. how many tasks were created in october, how many tasks were created in november, etc. (hint: use group by)
+SELECT monthname(created), count(*) AS number_of_tasks
+FROM task
+GROUP BY month(created)
+ORDER BY month(created);
```
Comment (flemeur) [inline databases/week2/homework-part-3.sql]: This one also gives me an error
```
Error Code: 1055. Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'hyf_lesson2.task.created' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```
Can you figure out why it gives an error?

File: databases/week2/homework-part-4.sql [hunk 6]
```diff
@@ -0,0 +1,44 @@
+SET NAMES utf8mb4;
```
Comment (flemeur) [inline databases/week2/homework-part-4.sql:1]: I think you forgot to add an E/R diagram for this part.

File: databases/week2/homework-part-4.sql [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+CREATE DATABASE library_inventory;
+USE library_inventory;
+
+CREATE table `author`(
+`id` INT UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+`name` varchar(255)
```
Comment (flemeur) [inline databases/week2/homework-part-4.sql]: I would make all the name columns NOT NULL

File: databases/week2/homework-part-4.sql [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+`rating` INT UNSIGNED NOT NULL,
+CONSTRAINT `fk_book_publisher` FOREIGN KEY (`publisher_id`) REFERENCES `publisher`(`id`) ON UPDATE CASCADE
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `book_genre`(
+`book_id` INT UNSIGNED NOT NULL,
+`genre_id` INT UNSIGNED NOT NULL,
+PRIMARY KEY(`book_id`, `genre_id`),
+CONSTRAINT `fk_book_genre_book` FOREIGN KEY(`book_id`) REFERENCES `book`(`id`) ON UPDATE CASCADE,
+CONSTRAINT `fk_book_genre_genre` FOREIGN KEY(`genre_id`) REFERENCES `genre`(`id`) ON UPDATE CASCADE
```
Comment (flemeur) [inline databases/week2/homework-part-4.sql]: It might make sense to have `ON DELETE CASCADE` on these many-to-many foreign key constraints

**Review by flemeur — COMMENTED** [review]
Hi @WENCHO22 Nice work! :+1: Here is my feedback

File: databases/week2/homework-part-3.sql (hunk 5 above)
Comment (WENCHO22) [reply databases/week2/homework-part-3.sql]: I implemented all the feedback but this one.
In my pc I get exactly what i suppose to get with no error at all, so I am a bit confused with this one..

Comment (flemeur) [reply databases/week2/homework-part-3.sql]: We mentioned this in the general homework feedback during the last lesson. Not sure if we explained it well enough though :)

Basically the error says that you should group by the thing that you want to select. 
So in your query you would either do

```sql
SELECT month(created), count(*) AS number_of_tasks
FROM task
GROUP BY month(created)
ORDER BY month(created);
```

or

```sql
SELECT monthname(created), count(*) AS number_of_tasks
FROM task
GROUP BY monthname(created)
ORDER BY monthname(created);
```

**Pull request merged**